src/inc/predeftlsslot.h

```diff
@@ -2,11 +2,14 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-
-
 #ifndef __PREDEFTLSSLOT_H__
 #define __PREDEFTLSSLOT_H__
 
+// ******************************************************************************
```

Comment:
There is a lot of cruft in this file. Saying that SOS depends on all of it sounds pretty unfortunate.

Can we abstract away the tiny bit of information that SOS depends on via DAC instead?

Reply:
SOS uses TlsIdx_ThreadType to get the flags (TlsThreadTypeFlag). The rest are just placeholders to make sure TlsIdx_ThreadType stays the same value across all versions of the runtime. The SOS works on 1.1, 2.0.x, 2.1.x, 2.2.x and 3.0.x.  I guess we could hard code this value to 11 to adds/removes won't affect SOS.

Yes, we could abstract the predeftlsslot.h/tls.h in the DAC, but SOS would still need the inc files and code explicitly for the older runtimes.  And we probably don't have enough time in 3.0 to do this new DAC interface.

/cc: @tommcdon, @noahfalk

Reply:
> I guess we could hard code this value to 11 to adds/removes won't affect SOS.

I would add `static_assert` that checks that the one value that SOS cares about stays intact, and reorder the other values around it. Number of the values are dead or are referenced by dead code (e.g. `TlsIdx_IceCap`).

> And we probably don't have enough time in 3.0 to do this new DAC interface.

I guess we can live with this for now. 

I am not sure whether `clrthreads -special` command is useful (I have not ever used it myself). The implementation of the command is Windows-specific. If the command is useful, we will want to re-implement it in platform-neutral way. And if it is not useful, we may just delete it.

Reply:
I'm going to ask around and see I can get a feel for really uses the clrthreads -special switch.  It does use a couple of runtime specific internal enums/offsets (tls.h and predeftlsslot.h) and only runs on Windows.  I like the idea of cutting it.

Reply:
I going to remove this -special option from Threads and the diagnostics repo's tls.h/predeftlsslot.h in a future PR. I removed the changes in the PR to those incs.